app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -82,8 +82,17 @@ def correct_puppet_environment
       end
 
       def content_and_puppet_matched?
```

Comment:
This is only called from `correct_puppet_environment` which is called in `before_save`, so this change shouldn't be needed.